backend/src/impl/db_models/system_metadata_model.py

```diff
@@ -140,7 +152,26 @@ def find(cls, page: int, page_size: int, system_name: Optional[str], task: Optio
                          {"creator": get_user().email}]
         cursor, total = super().find(filter, sort, page * page_size,
                                      page_size)
-        return SystemsReturn([cls.from_dict(doc) for doc in cursor], total)
+        documents = list(cursor)
+
+        # query datasets in batch to make it more efficient
+        dataset_ids: List[str] = []
+        for doc in documents:
+            if doc.get("dataset_metadata_id"):
+                dataset_ids.append(doc["dataset_metadata_id"])
+        datasets = DatasetMetaDataModel.find(0, 1000, dataset_ids).datasets
```

Comment:
Is 1000 just a large constant we arbitrarily set?

Reply:
Thanks! I added a `no_limit` option for `DatasetMetaDataModel.find()`. The code should be easier to follow now.

Reply:
Thanks! Looks great.